workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs

```diff
@@ -129,5 +130,214 @@ private static bool TryGetApkPath(string rootPath, out string apkPath)
+                }
+
+                if (!TryPrepareArguments(shouldConnectLocally, out var arguments))
+                {
+                    Debug.LogError("Failed to generate arguments");
+                    return;
+                }
+
+                // add SpatialOS arguments to iOS environment variables
+                var doc = new XmlDocument();
+                doc.Load(xcTestRunPath);
+                XmlNode spatialNode = null;
+                var rootNode = doc.DocumentElement.ChildNodes[0].ChildNodes[1];
+                for (var i = 0; i < rootNode.ChildNodes.Count; i++)
+                {
+                    var node = rootNode.ChildNodes[i];
+                    if (node.InnerText != "EnvironmentVariables")
+                    {
+                        continue;
+                    }
+
+                    var envValueNode = rootNode.ChildNodes[i + 1];
+                    for (var j = 0; j < envValueNode.ChildNodes.Count; j++)
+                    {
+                        var envNode = envValueNode.ChildNodes[j];
+                        if (envNode.InnerText != LaunchArguments.iOSEnvironmentKey)
+                        {
+                            continue;
+                        }
+
+                        spatialNode = node.ChildNodes[j + 1];
+                        break;
+                    }
+
+                    if (spatialNode != null)
+                    {
+                        spatialNode.InnerText = arguments;
+                    }
+                    else
+                    {
+                        var spatialKeyNode = doc.CreateNode("element", "key", string.Empty);
+                        spatialKeyNode.InnerText = LaunchArguments.iOSEnvironmentKey;
+                        var spatialValueNode = doc.CreateNode("element", "string", string.Empty);
+                        spatialValueNode.InnerText = arguments;
+                        envValueNode.AppendChild(spatialKeyNode);
+                        envValueNode.AppendChild(spatialValueNode);
+                    }
+
+                    break;
+                }
+
+                doc.Save(xcTestRunPath);
+
+                // fix document after updating it
+                var text = File.ReadAllText(xcTestRunPath);
+                text = text.Replace("[]>", ">");
+                File.WriteAllText(xcTestRunPath, text);
+
+                if (availableDevices.Count > 0)
+                {
+                    deviceUID = availableDevices[0].Trim('[', ']');
+                }
+                else
+                {
+                    // start simulator
+                    deviceUID = toolsConfig.GetSimulatorUID().Trim('[', ']');
+
+                    if (RedirectedProcess.Command("xcrun")
+                        .WithArgs("instruments", "-w", deviceUID, "-t", "Blank")
+                        .Run() != 0)
+                    {
+                        Debug.LogError("Was unable to start iOS Simulator.");
```

Comment:
Anything you can do to debug this at this point?

Reply:
you get error logs inside the Unity console, that should help you to debug it